feat: wire certificate table and page

flagrede opened the pull request:
## Description

Closes: regen-network/regen-registry#1726

- setup code and apollo config for indexer endpoint
- add `useFetchRetirements` and `useFetchRetirement` hooks
- wire certificate table
- wire certificate page

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-1953--regen-registry.netlify.app/ecocredits/portfolio
2. Click Retirement Certificates
3. Click View Certificate

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

flagrede commented:
@erikalogie you can have a look at the certificate table and page here: https://deploy-preview-1953--regen-registry.netlify.app/ecocredits/portfolio
Note that on the certificate page, the txHash is static for now while we add it to the indexer. Also, the indexer DB might be reset so if you're not seeing any retirements you can create new ones, they usually take only a few seconds to shop up.

erikalogie commented:
@flagrede it looks to me like the retirement certificates tab does not appear at certain user profile urls, i.e. this one: https://deploy-preview-1953--regen-registry.netlify.app/ecocredits/accounts/regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46/portfolio (Looks like this for me: https://monosnap.com/file/PV1AsgmoNo9Ynqd4q6oD4KIwnyBNeq)

Also, I think that we should change it so this button in the retire flow reads "view retirement certificates" https://monosnap.com/file/TFSZKzR9zhs5ZQSRDAha49Pw45rnuC and links to this new table. This could be part of a separate issue, but one thing I am concerned about is that the retirement table requires a refresh in order to see the newly retired credits in the table after retirement. It makes the user think that something went wrong. Can this be fixed as part of this pr or should it be separate?

flagrede commented:
@erikalogie 
>it looks to me like the retirement certificates tab does not appear at certain user profile urls, i.e. this one: https://deploy-preview-1953--regen-registry.netlify.app/ecocredits/accounts/regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46/portfolio (Looks like this for me: https://monosnap.com/file/PV1AsgmoNo9Ynqd4q6oD4KIwnyBNeq)

I wasn't aware we wanted to display the retirement table on the public profile page. I've changed it, it should appear on there as well now.

>Also, I think that we should change it so this button in the retire flow reads "view retirement certificates" https://monosnap.com/file/TFSZKzR9zhs5ZQSRDAha49Pw45rnuC and links to this new table. This could be part of a separate issue, but one thing I am concerned about is that the retirement table requires a refresh in order to see the newly retired credits in the table after retirement. It makes the user think that something went wrong. Can this be fixed as part of this pr or should it be separate?

I've updated both the button title and action to show the retirement table right after. Also the table is now being refreshed right away after clicking and also 5 seconds in case the block was not yet indexed.

erikalogie commented:
This is great! I think that we should also order the retirement certificates table such that the most recently retired appear at the top of the first page (so the reverse order from now).

blushi commented on web-registry/src/pages/Certificate/Certificate.utils.ts:
```diff
@@ -0,0 +1,58 @@
+    certificateTitle: CERTIFICATE_TITLE,
+    creditsUnits: Number(retirement.amountRetired),
+    creditUnitName: retirement.creditClassName,
+    equivalentTonsCO2: Number(retirement.amountRetired),
```
shouldn't we display this only if these are carbon credits?

flagrede replied:
Good catch, I actually prepare the data to do that but did not finish that part, will update it.

flagrede replied:
Fix in: [196e4ac](https://github.com/regen-network/regen-web/pull/1953/commits/196e4ac5d342ccfe5666623239c6ad187f92da80)

blushi reviewed (commented):
Code looks good, just came across something while testing:
I can't see the content of a given certificate page if not logged-in
I guess we might want this so if you share your certificate with someone, he can view it without signing in

erikalogie commented:
> Code looks good, just came across something while testing: I can't see the content of a given certificate page if not logged-in I guess we might want this so if you share your certificate with someone, he can view it without signing in

Oh yeah good call we definitely want these visible without being logged in

flagrede commented:
@erikalogie @blushi The page is now viewable even for not logged users.

blushi reviewed (commented):
We can now get the tx hash from the indexer db too

flagrede commented:
@blushi just wired the txHash.

blushi commented on web-registry/src/lib/normalizers/retirements/normalizeRetirement.types.ts:
```diff
@@ -0,0 +1,16 @@
+import { Maybe } from 'generated/graphql';
+import { Retirement } from 'generated/indexer-graphql';
+
+export type RetirementType = Maybe<
```
couldn't we just use `RetirementFieldsFragment` now instead?

blushi reviewed (approved):
tACK, nice work! pending a small nit
